Fix data properties

galamit86 opened the pull request:
This PR:

* Adds a step to the processing of datasets (`main`), where the `DataProperties` table column names are checked for any "." and replaced with "_". Ideally this would be done for all tables, but since there is no simple way to do this without loading the whole table to memory, this focuses on where this problem has actually occurred.
* Adds a function `fix_data_properties()` that:
  * downloads the `DataProperties` table from the latest GCS folder
  * renames the columns
  * uploads the renamed file to the same location, overwriting the exisitng file.

  thereby fixing the issue without having to re-upload the whole dataset.

galamit86 committed:
replace . in DataProperties tables

galamit86 committed:
add function to replace existing data properties tables

galamit86 committed:
simplify fix_data_properties

dkapitan commented:
@galamit86
You may want to consider including a more generic clean_names function where we could add more logic as we go along.

See https://gist.github.com/dkapitan/89ff20eeed38e6d9757fef9e09e23c3d

I have written than in the past to get rid of spaces. We could at a line to replace periods

galamit86 committed:
remove redundant code

galamit86 committed:
use generic function to clean name

galamit86 commented:
@dkapitan 
Added generic function based on yours, with 2 differences:
1. Added option to add other characters to replace (in this instance adding ".")
2. Return `s` instead of `s.lower()` - I tend to stick to the original names as far as possible. Additionally, since this is is not applied on all tables, but only on `DataProperties`, I think it's better not to lowercase only some tables.

dkapitan commented:
@galamit86
Looks good to me!
